Purchases/NSData+RCExtensions.m

```diff
@@ -0,0 +1,34 @@
+//
+// Created by Andrés Boedo on 3/4/20.
+// Copyright (c) 2020 Purchases. All rights reserved.
+//
+
+#import "NSData+RCExtensions.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+
+@interface NSData (RCExtensions)
+@end
+
+
+NS_ASSUME_NONNULL_END
+
+
+@implementation NSData (RCExtensions)
+
+- (NSString *)asString {
+    NSMutableString *deviceTokenString = [NSMutableString string];
+    [self enumerateByteRangesUsingBlock:^(const void *bytes,
+                                          NSRange byteRange,
+                                          BOOL *stop) {
+
+        for (NSUInteger i = 0; i < byteRange.length; ++i) {
+            [deviceTokenString appendFormat:@"%02x", ((uint8_t *) bytes)[i]];
+        }
+    }];
+    return deviceTokenString;
```

Comment:
So of NSData is nil, this will return "" right?

Reply:
well, if NSData is nil, this method will no-op because it's getting called on nil. 
if it's empty, it'll return an empty string. I'll add a test for it. 
We should be good either way

Reply:
actually, if it's empty, it still has bytes, so it'll return gibberish. But I can't imagine that you'd call this on empty data and expect a reasonable string. 
If you want to clear out the push token, the easiest is to send nil instead of empty data.